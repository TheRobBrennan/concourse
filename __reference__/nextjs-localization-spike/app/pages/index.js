import * as React from 'react'
import Link from 'next/link'
import { i18n, withTranslation } from '../i18n'
import Header from '../components/Header'
import Footer from '../components/Footer'

export const IndexPage = ({ t }) => (
  <React.Fragment>
    <main>
      <Header title={t('h1')} />
      <div>
        <button
          type="button"
          onClick={() => i18n.changeLanguage(i18n.language === 'en' ? 'de' : 'en')}
        >
          {t('change-locale')}
        </button>
        <Link href="/second-page">
          <button type="button">{t('to-second-page')}</button>
        </Link>
      </div>
    </main>
    <Footer />
  </React.Fragment>
)

IndexPage.getInitialProps = async () => ({
  namespacesRequired: ['common', 'footer'],
})

export default withTranslation('common')(IndexPage)
