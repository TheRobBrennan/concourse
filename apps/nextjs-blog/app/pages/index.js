import Head from "next/head"
import Layout, { siteTitle } from "../components/layout"
import utilStyles from "../styles/utils.module.css"

export default function Home() {
  return (
    <Layout home>
      <Head>
        <title>{siteTitle}</title>
      </Head>
      <section className={utilStyles.headingMd}>
        <p>
          Hi! I'm <strong>THE</strong> Rob Brennan. If you meet any others, they
          are imposters and must be...dealt with.
        </p>
      </section>
    </Layout>
  )
}
