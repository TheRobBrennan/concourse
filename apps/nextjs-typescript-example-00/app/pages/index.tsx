import { FunctionComponent } from "react"
import { GetStaticProps } from "next"
import Link from "next/link"
import Layout from "../components/Layout"

type IndexPageAPIResponse = {
  message: string
}

type IndexPageProps = {
  message?: string
}

const IndexPage: FunctionComponent<IndexPageProps> = ({ message }) => {
  return (
    <Layout title="Home | Next.js + TypeScript Example">
      <h1>Hello Next.js ðŸ‘‹</h1>
      <p>{message}</p>
      <p>
        <Link href="/about">
          <a>About</a>
        </Link>
      </p>
    </Layout>
  )
}

export const getStaticProps: GetStaticProps = async () => {
  try {
    const url = process.env.NEXT_PUBLIC_FLASK_API || ""
    const res = await fetch(url)
    const data: IndexPageAPIResponse = await res.json()
    const { message } = data

    if (!data) {
      return {
        notFound: true,
      }
    }

    return {
      // These props will be passed to the Next.js page component
      props: {
        message,
      },
    }
  } catch (err) {
    console.error(
      `Handling expected error when the Python Flask API is not available (e.g. building the Next.js container for development, etc)`
    )
    return {
      // These props will be passed to the Next.js page component
      props: {
        message: "",
      },
    }
  }
}

export default IndexPage
