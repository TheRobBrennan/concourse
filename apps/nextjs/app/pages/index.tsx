// import { GetStaticProps } from "next"
import Link from "next/link"
import { FunctionComponent } from "react"

import Layout from "../components/Layout"

// type IndexPageAPIResponse = {
//   message?: string
// }

type IndexPageProps = {
  message?: string
}

const IndexPage: FunctionComponent<IndexPageProps> = ({ message }) => {
  return (
    <Layout title="Home | Next.js + TypeScript Example">
      <h1>Hello Next.js ðŸ‘‹</h1>
      <p>{message}</p>
      <p>
        <Link href="/about">
          <a>About</a>
        </Link>
      </p>
    </Layout>
  )
}

// For examples using TypeScript for Next.js static generation and server-side rendering, refer to
// https://nextjs.org/docs/basic-features/typescript#static-generation-and-server-side-rendering
// export const getStaticProps: GetStaticProps = async () => {
//   try {
//     const url = process.env.NEXT_PUBLIC_EXTERNAL_API || ""
//     const res = await fetch(url)
//     const data: IndexPageAPIResponse = await res.json()
//     const { message } = data

//     if (!data) {
//       return {
//         notFound: true,
//       }
//     }

//     return {
//       // These props will be passed to the Next.js page component
//       props: {
//         message,
//       },
//     }
//   } catch (err) {
//     console.error(
//       `Handling expected error when the API is not available (e.g. building the Next.js container for development, etc)`
//     )

//     return {
//       // These props will be passed to the Next.js page component
//       props: {
//         message: "",
//       },
//     }
//   }
// }

export default IndexPage
