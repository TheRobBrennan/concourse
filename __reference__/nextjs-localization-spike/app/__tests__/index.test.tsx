import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { IndexPage } from '../pages/index'

describe('The default page', () => {
  // Mock our i18n library (see __reference__/nextjs-localization-spike/app/jest.setup.js)
  jest.mock('react-i18next', () => ({
    // this mock makes sure any components using the translate HoC receive the t function as a prop
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    withTranslation: () => (Component) => {
      Component.defaultProps = { ...Component.defaultProps, t: () => '' }
      return Component
    },
  }))

  // Define a mock translation function
  const mockTFunction = (key: string): string => {
    return key
  }

  it(`should render`, () => {
    const subject = shallow(<IndexPage t={mockTFunction} />)
    expect(subject).toBeDefined()
  })

  it('should match the snapshot', () => {
    const tree = renderer.create(<IndexPage t={mockTFunction} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
