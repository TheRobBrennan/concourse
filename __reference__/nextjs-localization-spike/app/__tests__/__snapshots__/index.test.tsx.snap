// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The default page should match the snapshot 1`] = `
Array [
  <main>
    <h1>
      h1
      <hr />
    </h1>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        change-locale
      </button>
      <button
        onClick={[Function]}
        onMouseEnter={[Function]}
        type="button"
      >
        to-second-page
      </button>
    </div>
  </main>,
  <footer>
    <p>
      description
    </p>
    <p>
      next-i18next v
      7.0.1
    </p>
  </footer>,
]
`;
